.github/workflows/golangci-lint.yml

```
@@ -0,0 +1,29 @@
+name: golangci-lint
+on:
+    push:
+        branches:
+            - main
+        paths:
+            - '**.go'
+    pull_request:
+        branches:
+            - main
+        paths:
+            - '**.go'
+
+permissions:
+    contents: read
+
+jobs:
+    golangci:
+        name: lint
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+            - uses: actions/setup-go@v5
+              with:
+                go-version: stable
+            - name: golangci-lint
+              uses: golangci/golangci-lint-action@v6
+              with:
+                version: v1.60
```

1. Give better names to the steps.
2. Use the same version of Golang as in `go.mod`.
3. Use the exact same version of golangci-lint as installed in the devcontainer's `Dockerfile`.
```suggestion
            - name: Check out code
              uses: actions/checkout@v4
            - name: Set up go
              uses: actions/setup-go@v5
              with:
                  go-version-file: "go.mod"
                  cache: true
            - name: Lint Golang
              uses: golangci/golangci-lint-action@v6
              with:
                  version: v1.60.3
```

---

@rlenglet, do you think we should take the version directly from Dockerfile(something like secrets.GOLANGCI_VERSION) instead of manually setting it to the one in dockerfile? Not sure how often we might change the GOLANGCI_VERSION

---

Not sure how to extract that from the Dockerfile dynamically in the GHA.
It's not worth the effort IMHO.
We can update that manually in two placed whenever needed.